src/extension/mapml/src/main/java/org/geoserver/mapml/MapMLDocumentBuilder.java

```diff
@@ -558,6 +562,10 @@ private MapMLLayerMetadata layerToMapMLLayerMetadata(RawLayer layer, String styl
             queryable = !layerGroupInfo.isQueryDisabled();
             layerName = layerGroupInfo.getName();
             layerTitle = getTitle(layerGroupInfo, layerName);
+            defaultMimeType =
+                    layerMeta.get(MapMLConstants.MAPML_MIME) != null
+                            ? layerMeta.get(MapMLConstants.MAPML_MIME).toString()
+                            : defaultMimeType;
```

Comment:
Changed to Optional.ofNullable format but still needs to appear in both parts of if because metadata is stored in resourceInfo in the case of LayerInfo and in the main metadata in the case of LayerGroupInfo.